auto/src/main/scala/besom/model/Names.scala

```diff
@@ -0,0 +1,124 @@
+package besom.model
+
+import scala.compiletime.*
+import scala.compiletime.ops.string.*
+import scala.language.implicitConversions
+
+/** Name is an identifier. */
+opaque type Name <: String = String
```

Comment:
this could probably be useful in core and codegen, but would require a separate module, just for model stuff